scm-dao-xml/src/main/java/sonia/scm/xml/XmlStreams.java

```diff
@@ -67,31 +62,13 @@ public static void close(XMLStreamReader reader) {
     }
   }
 
-  public static XMLStreamReader createReader(Path path) throws IOException, XMLStreamException {
-    return createReader(Files.newBufferedReader(path, StandardCharsets.UTF_8));
-  }
-
-  public static XMLStreamReader createReader(File file) throws IOException, XMLStreamException {
-    return createReader(file.toPath());
-  }
-
-  private static XMLStreamReader createReader(Reader reader) throws XMLStreamException {
+  public static XMLStreamReader createReader(Reader reader) throws XMLStreamException {
     XMLInputFactory factory = XMLInputFactory.newInstance();
     factory.setProperty(XMLInputFactory.IS_SUPPORTING_EXTERNAL_ENTITIES, Boolean.FALSE);
     return factory.createXMLStreamReader(reader);
   }
```

Comment:
If I understand this correctly, you are replacing the original call to `createReader(Path path)` so that you can close the inner buffered reader directly in the places where this is used. But as far as I can see, this should be done in the [`close()` function of `IndentXMLStreamWriter`](https://github.com/scm-manager/scm-manager/blob/develop/scm-core/src/main/java/sonia/scm/xml/IndentXMLStreamWriter.java#L80), too, or am I missing something here?

Reply:
@pfeuffer I didn't said anything about IndentXMLStreamWriter.
IndentXMLStreamWriter is just a wrapper around XMLStreamWriterImpl, which doesn't close given resource.
I found on githubs - before 2008 they did close resources, but then they decided leave that as is in order to:
- be consistent to other libraries, which do the same (without clarification who does the same)
- in order to keep opened InputStream, so may be user will want to read few entities/chunks from one xml.

So in this change I gave ability to caller side to decide - how to create the writer/reader and manager opened resource.

Reply:
@StNekroman Thanks, we see the issue here. We will try to fix this inside the `XMLStreams` util to make sure, that the streams will always be closed correctly without relying on the implementation all the time.